crates/rspack_plugin_split_chunks_new/src/plugin/module_group.rs

```diff
@@ -240,4 +308,70 @@ impl SplitChunksPlugin {
       module_group_map.remove(&key);
     });
   }
+
+  fn get_key<'a, I: Iterator<Item = &'a ChunkUkey>>(
+    chunks: I,
+    chunk_idx_map: &FxHashMap<ChunkUkey, ChunksKey>,
+  ) -> ChunksKey {
+    let mut result = ChunksKey::default();
+    for chunk in chunks {
+      let idx = chunk_idx_map.get(chunk).issue_expect();
+      result |= idx;
+    }
+    result
+  }
+
+  #[allow(clippy::type_complexity)]
+  fn prepare_combination_maps(
+    module_graph: &ModuleGraph,
+    chunk_graph: &ChunkGraph,
+    chunk_idx_map: &FxHashMap<ChunkUkey, ChunksKey>,
+  ) -> (
+    FxHashMap<ChunksKey, FxHashSet<ChunkUkey>>,
+    FxHashSet<ChunkUkey>,
+    FxHashMap<usize, Vec<FxHashSet<ChunkUkey>>>,
+  ) {
+    let mut chunk_sets_in_graph = FxHashMap::default();
+    let mut single_chunk_sets = FxHashSet::default();
+
+    for module in module_graph.modules().keys() {
+      let chunks = chunk_graph.get_module_chunks(*module);
+      let chunk_key = Self::get_key(chunks.iter(), chunk_idx_map);
+
+      if chunks.len() == 1 {
+        single_chunk_sets.insert(*chunks.iter().next().issue_expect());
+      } else {
+        chunk_sets_in_graph.insert(chunk_key, chunks.clone());
+      }
+    }
+
+    let mut chunk_sets_by_count = FxHashMap::<usize, Vec<FxHashSet<ChunkUkey>>>::default();
+
+    for chunks in chunk_sets_in_graph.values() {
+      let count = chunks.len();
+      chunk_sets_by_count
+        .entry(count)
+        .and_modify(|set| set.push(chunks.clone()))
+        .or_insert(Vec::new());
+    }
+
+    (chunk_sets_in_graph, single_chunk_sets, chunk_sets_by_count)
+  }
+}
+
+fn is_sub_set<T: std::hash::Hash + std::cmp::Eq>(
```

Comment:
You could use this API https://doc.rust-lang.org/std/collections/struct.HashSet.html#method.is_subset